docs: Reorganise examples (#2871)
nginxinc/kubernetes-ingress: docs/content/app-protect/configuration.md

---
title: Configuration

> description: "This document describes how to configure the NGINX App Protect WAF module."

Reviewer: Should the folder here also be `app-protect-waf` instead of just `app-protect`? Also the folder in `ingress-resources`?

I've also noticed that in the docs is `app-protect` but in the examples is `appprotect`.

---

This document describes how to configure the NGINX App Protect module
> Check out the complete NGINX Ingress Controller with App Protect example resources on GitHub [for VirtualServer resources](https://github.com/nginxinc/kubernetes-ingress/tree/v2.3.0/examples/custom-resources/waf) and [for Ingress resources](https://github.com/nginxinc/kubernetes-ingress/tree/v2.3.0/examples/ingress-resources/appprotect).

## Global Configuration

The NGINX Ingress Controller has a set of global configuration parameters that align with those available in the NGINX App Protect module. See [ConfigMap keys](/nginx-ingress-controller/configuration/global-configuration/configmap-resource/#modules) for the complete list. The App Protect parameters use the `app-protect*` prefix.

> > Check out the complete NGINX Ingress Controller with App Protect example resources on GitHub [for VirtualServer resources](https://github.com/nginxinc/kubernetes-ingress/tree/v2.3.0/examples/custom-resources/waf) and [for Ingress resources](https://github.com/nginxinc/kubernetes-ingress/tree/v2.3.0/examples/ingress-resources/appprotect).

Reviewer: I'm adding the comment here, but it's general, should we rename the folders to `appprotect-dos` and `approtect-waf` since we're at it? 

Also, move from calling it just App Protect to App Protect WAF? 

cc @nginxinc/nginx-docs

Reviewer 2: @lucacome agreed on both counts

Reviewer 3: yes, absolutely.

Author: Will do 👍🏼

Reviewer 2: One thing to take into consideration here is that the file path == the path to the doc in the docs website, so if you change the file path you break links and bookmarks. 

You can add the previous path to the frontmatter if you do move docs around, so Hugo will automatically redirect from the old path to the new path. 
https://gohugo.io/content-management/urls/#aliases

The request will be blocked.

The `link` option is also available in the `openApiFileReference` property and is synonymous with the `open-api-files` property as seen in the App Protect WAF policy example above.

**Note**: `openApiFileReference` is not an array.

## Configuration in NGINX Plus Ingress Controller using Virtual Server Resource
In this example we deploy the NGINX Plus Ingress Controller with NGINX App Protect WAF, a simple web application and then configure load balancing and WAF protection for that application using the VirtualServer resource.

> **Note:** This example, and the files referenced, can be found [here](https://github.com/nginxinc/kubernetes-ingress/tree/v2.3.0/examples/custom-resources/waf).

Reviewer 2: ```suggestion
**Note:** You can find the example, and the files referenced, on [GitHub](https://github.com/nginxinc/kubernetes-ingress/tree/v2.3.0/examples/custom-resources/waf).
```

   $ kubectl apply -f https://raw.githubusercontent.com/nginxinc/kubernetes-ingress/v2.3.0/examples/custom-resources/waf/syslog.yaml
   ```

> 2. Create the User Defined Signature, App Protect WAF policy and log configuration:

Reviewer 2: ```suggestion
2. Create the User Defined Signature, App Protect WAF policy, and log configuration:
```
